solutions/devsprint-bruno-faganello-1/FinanceApp/Network/HTTPClient.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  HTTPClient.swift
+//  FinanceApp
+//
+//  Created by Victor Catão on 17/03/22.
+//
+
+import Foundation
+
+protocol HTTPClient {
+    func sendRequest<T: Decodable>(endpoint: Endpoint, responseModel: T.Type) async -> Result<T, RequestError>
+}
+
+extension HTTPClient {
+    func sendRequest<T: Decodable>(
+        endpoint: Endpoint,
+        responseModel: T.Type
+    ) async -> Result<T, RequestError> {
+        guard let url = URL(string: endpoint.baseURL + endpoint.path) else {
```

Comment:
[Question]
If your endpoint path has a space or special char this will broke.
You can use `URLComponents` to make this safer